source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc

```diff
@@ -162,6 +162,10 @@ void SimpleHttpCache::updateHeaders(const LookupContext& lookup_context,
         if (headersNotToUpdate().contains(lower_case_key)) {
           return Http::HeaderMap::Iterate::Continue;
         }
+        // `setCopy` is currently implemented as `remove` then `addCopy`. But it is not used here
+        // because (1) if there's a repeated header in the incoming update, the second occurrence 
+        // would overwrite our first occurrence. (2) The behaviour of `setCopy` is implementation
+        // dependent. A different implementation might only replace set first occurrence.
```

Comment:
Suggest:
```
// The behavior of setCopy is different between built-in O(1) headers and arbitrary headers. With
// the latter, only the first occurrence is changed, which is not what we need here; we always want
// to replace all the occurrences. Thus we use `remove` and then `addCopy`.
```

Reply:
Sorry, I think I didn't explain the state of `setCopy` well. For both O(1) headers and arbitrary headers, all occurrences would be replaced by exactly one new value. But whether it actually does that is dependent on how `HeaderMapImpl` implements it. I think I'll add this part of the suggestion
```
// we always want to replace all the occurrences. Thus we use `remove` and then `addCopy`.
```